src/bootstrap/src/utils/cc_detect.rs

```diff
@@ -172,6 +189,22 @@ pub fn find_target(build: &Build, target: TargetSelection) {
     }
 }
 
+/// Determines the default compiler for a given target and language when not explicitly
+/// configured in `config.toml`.
+///
+/// This function handles special cases for certain targets:
```

Comment:
This explicit list for each case seems kind of overly much. The first sentence was enough: "Determines the default compiler for a given target"